pandas/io/json/_json.py

```diff
@@ -1221,7 +1222,7 @@ def _try_convert_data(
         if len(data) and data.dtype in ("float", "object"):
             # coerce ints if we can
             try:
-                new_data = data.astype("int64")
+                new_data = backup_data.astype("int64")
```

Comment:
Instead of this, it seems to me we should first try to convert the data to integer on L1206 above, then if that fails try float. That way we aren't converting to float then back to int unnecessarily.